subsys/console/Kconfig

```diff
@@ -11,12 +11,10 @@ if CONSOLE_SUBSYS
 choice
 	prompt "Console 'get' function selection"
 	optional
-	depends on UART_CONSOLE && SERIAL_SUPPORT_INTERRUPT
+	depends on UART_CONSOLE
 
 config CONSOLE_GETCHAR
 	bool "Character by character input and output"
-	select UART_CONSOLE_DEBUG_SERVER_HOOKS
```

Comment:
Why are these being removed? Did you verify that all subsys/console-using software (including out of tree) keeps working?

Reply:
> Why are these being removed? 

If I remember correctly it was due it only brings the requirement to use only interrupt driven drivers.

>Did you verify that all subsys/console-using software (including out of tree) keeps working?

C'mon Paul what's that ? Regressions was ok.

Reply:
> If I remember correctly...

And that's exactly my point - please don't make yourself, me, and any other who will deal with this code coming years to guess. And my remembering is a bit different anyway - they were there because subsys/console reuses some parts of old hacked-up UART code, and those were the options needed to select that code to be available.

> C'mon Paul what's that ?

That was a question. That's how reviews are done - reviews ask questions and provide suggestions to the patch author. In this PR, the suggestion is for more detailed commit message, and question regarding testing status. I might be able to help with testing if you can't do it yourself (sorry in advance if that may take extra time).

> Regressions was ok.

Sorry, I'm not sure I understand this phrase, can you elaborate?